pkg/endpoint/endpoint.go

```diff
@@ -236,39 +236,69 @@ func (e *Endpoint) GetModel() *models.Endpoint {
 	}
 }
 
-type statusLog struct {
+// statusLogMsg represents a log message.
+type statusLogMsg struct {
 	Status    Status    `json:"status"`
 	Timestamp time.Time `json:"timestamp"`
 }
 
-type EndpointStatus struct {
-	Log     []*statusLog `json:"log,omitempty"`
-	Index   int          `json:"index"`
-	indexMU sync.RWMutex
+// statusLog represents a slice of statusLogMsg.
+type statusLog []*statusLogMsg
+
+// priorityStatus represents a map of a single statusLogMsg by StatusPriority.
+type priorityStatus map[StatusPriority]*statusLogMsg
```

Comment:
I think you want a struct `ComponentStatus` which is a member `Latest` in `EndpointStatus`. I know what you mean by priority but it's misleading. I think you really mean the latest status.